core/labeled-replies-3.3.md

```diff
@@ -0,0 +1,132 @@
+---
+title: Labeled responses
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: "Alexey Sokolov"
+    period: "2015"
+    email: "alexey-irc@asokolov.org"
+  -
+    name: "James Wheare"
+    period: "2016"
+    email: "james@irccloud.com"
+---
+
+## Notes for implementing work-in-progress version
+
+This is a work-in-progress specification.
+
+Software implementing this work-in-progress specification MUST NOT use the
+unprefixed `label` tag name. Instead, implementations SHOULD use
+the `draft/label` tag name to be interoperable with other software
+implementing a compatible work-in-progress version.
+
+The final version of the specification will use an unprefixed tag name.
+
+## Introduction
+
+This specification adds a new message tag sent by clients and repeated by servers to correlate responses with a specific request.
+
+## Motivation
+
+Certain client actions can result in responses from the server that vary in interpretion depending on how they were triggered, or otherwise lack a robust way to correlate with local state. Clients have historically needed to keep track of additional local state and/or apply comparison heuristics to server responses to correlate these appropriately.
+
+Labeled responses enable a much simpler form of correlation by using a single id attached to a client request and repeated by the server in its response.
+
+Additionally, labeled responses allow bouncers with multiple connected clients to direct responses (such as WHOIS queries or error messages, see examples) to the correct recipient.
+
+## Architecture
+
+### Dependencies
+
+This specification uses the [message tags framework](/specs/core/message-tags-3.2.html) and relies on support for the [`batch`](/specs/extensions/batch-3.2.html) capability which MUST be negotiated at the same time.
+
+### Capabilities
+
+This speciciation adds the `draft/labeled-response` capability.
+
+Clients requesting this capability indicate that they are capable of handling the message tag described below from servers.
+
+Servers advertising this capability indicate that they are capable of handling the message tag described below from clients, and will use with the same tag and value in their response.
+
+Servers MUST reject the capability at negotiation time if the `batch` capability is not also negotiated. Additionally, servers supporting `cap-notify` MUST cancel the capability if `batch` is removed at any time.
+
+### Batch types
+
+This specification adds the `draft/labeled-response` batch type, described below.
+
+### Tags
+
+This specification adds the `draft/label` message tag, which has one required value.
```

Comment:
Could clarify this, tags either have a value not have one but the wording here implies there may be multiple values for a tag